File: module/app/signalJ/SignalJGlobal.scala
package signalJ

import play.GlobalSettings
import play.api.Play
import play.api.mvc.Handler
import play.mvc.Http

/**
 * Can be used by the app to enable ServerSentEvents or LongPolling
 *
 * Another option if you already have a Global in your project, is to call TransportTransformer in your own Global
 */
class SignalJGlobal extends GlobalSettings {
  /**
   * Reroute requests with paths like "/connect?transport=webSockets" to "/connect/webSockets".
   */
  override def onRouteRequest(request: Http.RequestHeader): Handler = {
    val newRequest = TransportTransformer.transform(request)
      (Play.maybeApplication flatMap { app =>
        app.routes flatMap { router =>
          router.handlerFor(newRequest)
        }
      }).orNull
  }
}

File: module/app/signalJ/SignalJGlobal.java
package signalJ;

import play.GlobalSettings;
import play.Logger;
import play.api.mvc.Handler;
import play.mvc.Http;

public class SignalJGlobal extends GlobalSettings {
    @Override
    public Handler onRouteRequest(Http.RequestHeader requestHeader) {
        Logger.info("Uri:" + requestHeader.uri());
        final Http.RequestHeader newHeader = JavaTransportTransformer.Transform(requestHeader);
        if(!requestHeader.uri().equalsIgnoreCase(newHeader.uri())) Logger.info("New Uri:" + newHeader.uri());
        return super.onRouteRequest(newHeader);
    }
}